### PR TITLE
Add GCP credentials to the trainer node

### PR DESCRIPTION
**Pull request by anayden**

### CODE REVIEW

File: Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile-trainer [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
 
+# Include GCP service account credentials with the build
+# Run docker build .... --build-arg GCP_CREDS_JSON_BASE64="xxxx"
+# where xxxx is a base64-encoded version of GCP creds JSON
+# The lab repo has a secret named GCP_CREDS_JSON_BASE64 with required data
+ARG GCP_CREDS_JSON_BASE64
+ENV GCP_CREDS_JSON_BASE64_ENV=${GCP_CREDS_JSON_BASE64}
+ENV GOOGLE_APPLICATION_CREDENTIALS=/root/gcp.json
+RUN echo "${GCP_CREDS_JSON_BASE64_ENV}"| base64 --decode > "${GOOGLE_APPLICATION_CREDENTIALS}"
+
```
Comment (savagej) [inline Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile-trainer:11]: This is amazing!!!

File: Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile-trainer [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
 RUN pip install --upgrade pip && \
     pip install torch==1.7.1+cpu torchvision==0.8.2+cpu torchaudio==0.7.2 torchserve torch-model-archiver -f https://download.pytorch.org/whl/torch_stable.html && \
```
Comment (savagej) [inline Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile-trainer:13]: I don't think this line is needed as `mlflow run` will use the included conda.yaml to build an environment then train.
Might be better in a different PR though.

File: Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile-trainer [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
-    pip install --no-cache-dir numpy mlflow transformers==3.5.0 sklearn tqdm torchtext pandas cloudpickle boto3
+    pip install --no-cache-dir numpy mlflow transformers==3.5.0 sklearn tqdm torchtext pandas cloudpickle boto3 google-cloud-storage
```
Comment (savagej) [inline Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile-trainer:14]: Similar to comment above I don't think we need anything except mlflow and google-cloud-storage. Haven't checked this though.

**Review by savagej — APPROVED** [review]
I think my comments above are better fixed in a new PR (if they even need fixing)

**Comment by anayden** [comment]
@savagej thanks for your review. Your comments re the rest of the Dockerfile look good and I agree they're worth another PR :)

File: Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile-trainer (hunk 2 above)
Comment (anayden) [reply Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile-trainer:13]: @savagej I tested your suggestion and it didn't work :(
mlflow either uses conda env OR docker env, so if we use docker we have to package it with all dependencies
